frontend/src/util/SmartAuth.js

```diff
@@ -0,0 +1,73 @@
+import { useLocation } from "react-router-dom";
+import { Login } from "../pages/Login";
+import Cookies from 'js-cookie';
+
+export class SmartAuth {
+
+    static #citizen = null;
+
+    static async #fetchData(token) {
+        const url = `http://auth.smartcityproject.net:8080/verify`;
+        fetch(url, {
+            method: 'POST',
+            body: encodeURIComponent('code') + '=' + encodeURIComponent(token),
+            headers: { 'Content-Type': 'application/x-www-form-urlencoded' }
+        })
+            .then(response => response.json())
```

Comment:
Hier wirst du n Problem bekommen, wenn der Token nicht korrekt sein sollte, bzw. SmartAuth nicht die Session findet, wenn die abgelaufen ist oder so. response.json() wirft dann den Fehler "Unexpected end of JSON input". Daran kannst du jetzt erstmal nix ändern. Da muss ich mit Jonathan sprechen, dass er das ändert. Was du hier aber machen musst, ist prüfen, ob der User wirklich verifiziert ist. Wenn die Session nicht aufgelöst werden kann, dann bekommst du da ein 404 zurück. Das musst du auch irgendwie hier mit einbinden, weil wie gesagt keine Userdaten geschickt werden, sondern irgendein Error.